### PR TITLE
Add base linter test

### LINKED ISSUE #2: Add basic tests
As this repo will mostly contain Terraform files, we could have a basic test that runs Terraform linter.

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

Fixes #2

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Do we need to check terraform and terragrunt code format by `terraform fmt` and `terragrunt hclfmt`?

/assign @irbekrm

**Comment by irbekrm** [comment]
Thank you for the PR @lonelyCZ !

> Do we need to check terraform and terragrunt code format by terraform fmt and terragrunt hclfmt?

That's a good question. I see that this PR includes some changes from `terraform fmt` and `terraform validate` would not catch that.
I think that the test should not itself make changes using `terraform fmt`, but perhaps the test could check if there are no changes if `terraform fmt` is run? i.e by running the `terraform fmt -diff -recursive` and checking that there is no diff output. This would be similar to what we do in cert-manager https://github.com/cert-manager/cert-manager/blob/master/hack/verify-deps.sh#L65

What do you think?

Otherwise this looks good, next thing could be to add a CI test that runs this test https://github.com/cert-manager/testing-addons/issues/3

**Comment by lonelyCZ** [comment]
The test is below

```
[root@master68 testing-addons]# scripts/verify-staticcheck.sh
cm-vault/common.tfvars
--- old/cm-vault/common.tfvars
+++ new/cm-vault/common.tfvars
@@ -1,3 +1,3 @@

 # Path to the kubeconfig file to use for connecting kubernetes cluster. Default is "~/.kube/config"
-kubeconfig_path="~/.kube/config"
+kubeconfig_path = "~/.kube/config"
cm-vault/vault-install/main.tf
--- old/cm-vault/vault-install/main.tf
+++ new/cm-vault/vault-install/main.tf
@@ -35,10 +35,10 @@
 # ready https://github.com/helm/helm/pull/10920
 resource "null_resource" "wait_for_vault" {
   provisioner "local-exec" {
-   environment = {
-    KUBECONFIG = pathexpand(var.kubeconfig_path)
-   }
-   command = "kubectl wait --for condition=Ready=true pod/vault-0 -n vault --timeout 300s"
+    environment = {
+      KUBECONFIG = pathexpand(var.kubeconfig_path)
+    }
+    command = "kubectl wait --for condition=Ready=true pod/vault-0 -n vault --timeout 300s"
   }
   depends_on = [
     helm_release.vault

Please run "terraform fmt -recursive" to format the codes.
[root@master68 testing-addons]# terraform fmt -recursive
cm-vault/common.tfvars
cm-vault/vault-install/main.tf
[root@master68 testing-addons]# scripts/verify-staticcheck.sh
ERRO[0000] 1 error occurred:
        * Invalid file format /root/lonelyCZ/testing-addons/cm-vault/vault-config/terragrunt.hcl

ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1

Please run "terragrunt hclfmt" to format the codes.
[root@master68 testing-addons]# terragrunt hclfmt
INFO[0000] /root/lonelyCZ/testing-addons/cm-vault/vault-config/terragrunt.hcl was updated
[root@master68 testing-addons]# scripts/verify-staticcheck.sh
INFO[0000] The stack at /root/lonelyCZ/testing-addons will be processed in the following order for command init:
Group 1
- Module /root/lonelyCZ/testing-addons/cm-vault/cm-install
- Module /root/lonelyCZ/testing-addons/cm-vault/vault-install

Group 2
- Module /root/lonelyCZ/testing-addons/cm-vault/vault-config

Group 3
- Module /root/lonelyCZ/testing-addons/cm-vault/cm-config


Initializing the backend...

Initializing the backend...

Initializing provider plugins...
- Reusing previous version of hashicorp/helm from the dependency lock file

Initializing provider plugins...
- Reusing previous version of hashicorp/helm from the dependency lock file
- Reusing previous version of hashicorp/null from the dependency lock file
- Using previously-installed hashicorp/helm v2.5.1

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
- Using previously-installed hashicorp/helm v2.5.1
- Using previously-installed hashicorp/null v3.1.1

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Initializing the backend...

Initializing provider plugins...
- Reusing previous version of hashicorp/vault from the dependency lock file
- Using previously-installed hashicorp/vault v3.5.0

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Initializing the backend...

Initializing provider plugins...
- Reusing previous version of hashicorp/kubernetes from the dependency lock file
- Using previously-installed hashicorp/kubernetes v2.11.0

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
INFO[0000] The stack at /root/lonelyCZ/testing-addons will be processed in the following order for command validate:
Group 1
- Module /root/lonelyCZ/testing-addons/cm-vault/cm-install
- Module /root/lonelyCZ/testing-addons/cm-vault/vault-install

Group 2
- Module /root/lonelyCZ/testing-addons/cm-vault/vault-config

Group 3
- Module /root/lonelyCZ/testing-addons/cm-vault/cm-config

Success! The configuration is valid.

Success! The configuration is valid.

Success! The configuration is valid.

Success! The configuration is valid.

Congratulations! All Terraform source files have passed staticcheck.
```

**Comment by lonelyCZ** [comment]
Thanks your suggestions, please review it again.

/assign @irbekrm

File: scripts/verify-staticcheck.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  echo
+  echo 'Please run "terragrunt hclfmt" to format the codes.'
+  exit 1
+fi
+
+# Check whether the configuration is valid 
+terragrunt run-all init
+
+if terragrunt run-all validate; then
+  echo 'Congratulations! All Terraform source files have passed staticcheck.'
```
Comment (irbekrm) [inline scripts/verify-staticcheck.sh:26]: Thanks for adding this! I think the positive success messages are nice to see 👍🏼

File: scripts/verify-staticcheck.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}"
+
+# Check terraform codes format
+if ! terraform fmt -recursive -diff -check; then
```
Comment (irbekrm) [inline scripts/verify-staticcheck.sh:9]: The `-check` flag is a good find 👍🏼

**Review by irbekrm — APPROVED** [review]
Thank you @lonelyCZ !

I've tested the script and saw that it now picks up incorrect formatting!

/lgtm

**Pull request merged**